mlos_bench/mlos_bench/tests/services/remote/mock/mock_fileshare_service.py

```diff
@@ -24,12 +24,10 @@ class MockFileShareService(FileShareService, SupportsFileShareOps):
     def __init__(self, config: Optional[Dict[str, Any]] = None,
                  global_config: Optional[Dict[str, Any]] = None,
                  parent: Optional[Service] = None):
+        # IMPORTANT: Save the local methods before invoking the base class constructor
+        local_methods = [self.upload, self.download]
         super().__init__(config, global_config, parent)
-
-        self.register([
-            self.download,
-            self.upload,
-        ])
+        self.register(local_methods)
```

Comment:
This pattern isn't followed in the `AzureFileshare` service.  Does it need to be?
The base `FileShareService` class does this, but I'm not sure off hand if it's registering the abstract methods, or the subclasses methods.  If it's already happening in the base class, can we drop it here?

Reply:
I don't remember whether it was a bug or a feature :) I think it would be safer to enforce registering of the local methods in every class even if the superclass already does that